tests/integration/ha/test_ha.py

```diff
@@ -249,6 +252,61 @@ async def test_freeze_broker_with_topic_leader(
     assert_continuous_writes_consistency(result=result)
 
 
+@pytest.mark.abort_on_fail
+async def test_network_cut_without_ip_change(
+    ops_test: OpsTest,
+    c_writes: ContinuousWrites,
+    c_writes_runner: ContinuousWrites,
+):
+    # Let some time pass for messages to be produced
+    await asyncio.sleep(10)
+
+    topic_description = await get_topic_description(
+        ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME
+    )
+    initial_leader_num = topic_description.leader
+    leader_machine_name = await get_unit_machine_name(
+        ops_test=ops_test, unit_name=f"{APP_NAME}/{initial_leader_num}"
+    )
+
+    logger.info(
+        f"Throttling network for leader of topic '{ContinuousWrites.TOPIC_NAME}': {initial_leader_num}"
+    )
+    network_throttle(machine_name=leader_machine_name)
+    await asyncio.sleep(REELECTION_TIME * 2)
+
+    # verify replica is not in sync
+    topic_description = await get_topic_description(
+        ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME
+    )
+    assert topic_description.in_sync_replicas == {0, 1, 2} - {initial_leader_num}
```

Comment:
**todo:** Let's try to avoid hard-coding unit numbers here. 

If (for whatever reason) we add a test to redeploy the cluster, these numbers will need to be changed. Better to generate all unit integers from `ops_test.model.applications[APP_NAME].units` and remove it there I think.

Reply:
This one and the suggestion below will go on their own PR I think, as it affects all tests, not just this one.

Reply:
I agree with @zmraul as that would then go beyond the scope of the PR. We could just add a separate ticket (tech debt for a small refactoring), I believe Marc suggestion is sensible. Would you mind of creating this ticket please?